va/va.c

```diff
@@ -2432,3 +2432,44 @@ VAStatus vaProtectedSessionExecute(
     return status;
 }
 
+VAStatus vaQueryDeviceAttributes(
+     VADisplay dpy,
+     VADeviceAttrib *attrib_list,
+     uint32_t       *num_attribs
+)
+{
+  VAStatus vaStatus = VA_STATUS_SUCCESS;
+  VADriverContextP ctx;
+  CHECK_DISPLAY(dpy);
+  ctx = CTX(dpy);
+
+  if(ctx->vtable->vaQueryDeviceAttributes)
+  {
+    vaStatus = ctx->vtable->vaQueryDeviceAttributes(ctx, attrib_list, num_attribs);
+  }
+  else
+  {
+    vaStatus = VA_STATUS_ERROR_UNIMPLEMENTED;
+  }
+
+  return vaStatus;
+}
+
+VAStatus vaSetDeviceAttributes(
+     VADisplay dpy,
+     VADeviceAttrib *attrib_list,
```

Comment:
Initial concept for this API was slightly different, in particular consideration was to have attribs settable before vaInitialize. In this sense old API did not work for us... But then concept has changed we forgot to evaluate the new API vs. existing one.

@XinfengZhang : indeed, 2 sets of APIs start to look weird. Should we converge and just declare couple new display attributes?

Reply:
I wonder whether display attributes is suitable , because current display attributes seems just related with "Display" not adaptor

Reply:
I think yes, considering that adaptor/display is kind of merged thing in libva, and I would say even on Linux device exposure level (since we can use /dev/dri/cardN for display and encoding). So, having all attributes together seems ok to me - they just configure different aspects of the same entity.